tests/catalog/test_db.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+from sqlalchemy import create_engine
+from sqlalchemy.orm import Session
+from datetime import datetime
+from recap.catalogs.db import DatabaseCatalog, CatalogEntry
+
+class TestCatalogEntry:
+    def test_is_deleted(self):
+        entry = CatalogEntry(deleted_at=datetime.now())
+        assert entry.is_deleted() == True
+
+        entry = CatalogEntry(deleted_at=None)
+        assert entry.is_deleted() == False
+
+class TestDatabaseCatalog:
+    @pytest.fixture
+    def engine(self):
+        return create_engine('sqlite:///:memory:')
+
+    @pytest.fixture
+    def catalog(self, engine):
+        return DatabaseCatalog(engine)
+
+    def test_catalog_init(self, catalog):
+        assert isinstance(catalog.Session(), Session)
+
+    def test_catalog_touch(self, catalog):
+        # Test touching a path that doesn't exist
+        catalog.touch("/databases/table")
+        with catalog.Session() as session:
```

Comment:
Rather than sneaking into the `.Session()` instance variable, can you use `ls()` here to see if the child was created? It's a bit safer not to depend on internal implementation details for the test.